### PR TITLE
Switch to Java 11 to use the built-in HttpClient instead of OkHttpClient

### PR DESCRIPTION
**Pull request by mrts**
WE2-790

### CODE REVIEW

**Comment by angryziber** [comment]
Looks good to me.
Rethrowing InterruptedException is suspicious, but probably ok to preserve the interface signature.

**Review by angryziber — APPROVED** [review]
Looks good to me.
Rethrowing InterruptedException is suspicious, but probably ok to preserve the interface signature.

**Comment by angryziber** [comment]
Maybe class name could be improved to something like `OcspClientImpl`, because it's the only one and uses only built-in JDK dependencies.

**Comment by mrts** [comment]
Thanks! 

Yes, exposing `InterruptedException` in the interface signature would require that the developers know that they have to call `Thread.currentThread().interrupt()` before rethrowing it. I think this puts too large onus on them and may cause problems, so it is better to catch it, handle the interruption and rethrow as `IOException` ourselves. The assumption is that the interruption is a rare event and developers want to treat it like any other I/O failure.

I agree, `OcspClientImpl` is better considering that it is the only one and uses only built-in JDK dependencies. `BuiltinHttpClientOcspClient` is precise, but hard to read and justified only in case there are many implementations.

File: src/main/java/eu/webeid/security/validator/ocsp/OcspClientImpl.java [hunk 1]
```diff
@@ -66,31 +62,36 @@ public static OcspClient build(Duration ocspRequestTimeout) {
      */
     @Override
     public OCSPResp request(URI uri, OCSPReq ocspReq) throws IOException {
-        final RequestBody requestBody = RequestBody.create(ocspReq.getEncoded(), OCSP_REQUEST_TYPE);
-        final Request request = new Request.Builder()
-            .url(uri.toURL())
-            .post(requestBody)
+        final HttpRequest request = HttpRequest.newBuilder()
+            .uri(uri)
+            .header("Content-Type", OCSP_REQUEST_TYPE)
```
Comment (metsma) [inline src/main/java/eu/webeid/security/validator/ocsp/OcspClientImpl.java:67]: Can you use here also static variable?

File: src/main/java/eu/webeid/security/validator/ocsp/OcspClientImpl.java [hunk 2]
```diff
@@ -66,31 +62,36 @@ public static OcspClient build(Duration ocspRequestTimeout) {
...
+            Thread.currentThread().interrupt();
+            throw new IOException("Interrupted while sending OCSP request", e);
         }
+
+        if (response.statusCode() != 200) {
+            throw new IOException("OCSP request was not successful, response: " + response);
+        } else {
+            LOG.debug("OCSP response: {}", response);
+        }
+        final String contentType = response.headers().firstValue("Content-Type").orElse("");
```
Comment (metsma) [inline src/main/java/eu/webeid/security/validator/ocsp/OcspClientImpl.java:85]: Same here

**Review by metsma — CHANGES_REQUESTED** [review]

File: src/main/java/eu/webeid/security/validator/ocsp/OcspClientImpl.java (hunk 1 above)
Comment (mrts) [reply src/main/java/eu/webeid/security/validator/ocsp/OcspClientImpl.java:67]: Good point, thanks! I will add the static variable.

Comment (mrts) [reply src/main/java/eu/webeid/security/validator/ocsp/OcspClientImpl.java]: Done.

**Review by metsma — APPROVED** [review]

**Pull request merged**